File: stuff/tag/2021_03_15/commonFunctions/callElements.m
function dataFE = callElements(obj,setupObject,flag)
dataFE = [];
for index1 = 1:obj.numberOfContinuumObjects
    continuumObj = evalin('base',obj.listContinuumObjects{index1});
     for index2 = 1:numel(continuumObj)
         if continuumObj(index2).flagNewton.(flag)
            elementName = strcat(continuumObj(index2).elementDisplacementType,continuumObj(index2).materialName,setupObject.integrator);
            dataFEindex = feval(elementName,continuumObj(index2),setupObject);
            dataFE = [dataFE, dataFEindex];
         end
     end
end

File: stuff/pyfemkit/commonFunctions/callElements.py
import numpy as np
from continuumClasses.solidClass.displacementHookeEndpoint import displacementHookeEndpoint
from continuumClasses.solidClass.displacementHookeAutoDiffEndpoint import displacementHookeAutoDiffEndpoint
from continuumClasses.solidClass.displacementParallelHookeAutoDiffEndpoint import displacementParallelHookeAutoDiffEndpoint
from continuumClasses.solidClass.displacementSaintVenantEndpoint import displacementSaintVenantEndpoint
from continuumClasses.solidClass.displacementSaintVenantAutoDiffEndpoint import displacementSaintVenantAutoDiffEndpoint
from continuumClasses.solidClass.displacementSCMooneyRivlinDiscreteGradient import displacementSCMooneyRivlinDiscreteGradient
from continuumClasses.solidClass.displacementSCMooneyRivlinAutoDiffDiscreteGradient import displacementSCMooneyRivlinAutoDiffDiscreteGradient
from continuumClasses.solidClass.displacementParallelHookeEndpoint import displacementParallelHookeEndpoint

def callElements(self,setupObject,flag):
    dataFE = []
    for index1 in range(self.numberOfContinuumObjects):
        continuumObject = self.listContinuumObjects[index1]
        if continuumObject.flagNewton[flag]:
            elementName = continuumObject.elementDisplacementType + continuumObject.materialName + setupObject.integrator
            dataFEindex = eval(elementName)(continuumObject,setupObject)
            # FIXME
            # dataFE = [dataFE, dataFEindex]
            dataFE = dataFEindex
    return dataFE